### PR TITLE
parsers for Date and UUID types

### PR DESCRIPTION
**Pull request by eklavya**
for capturing path variables.

### CODE REVIEW

File: core/src/test/scala/org/http4s/rho/bits/StringParserSpec.scala [hunk 1]
```diff
@@ -51,5 +54,15 @@ class StringParserSpec extends Specification {
     "parse non-short" in {
       new ShortParser[IO]().parse("12345678") must haveClass[FailureResponse[IO]]
     }
+
+    "parse date" in {
```
Comment (jcranky) [inline core/src/test/scala/org/http4s/rho/bits/StringParserSpec.scala]: Should we test the failure cases as well ?

File: core/src/main/scala/org/http4s/rho/bits/StringParser.scala [hunk 2]
```diff
@@ -82,6 +114,8 @@ object StringParser {
   implicit def intParser[F[_]]: IntParser[F] = new IntParser[F]()
   implicit def longParser[F[_]]: LongParser[F] = new LongParser[F]()
   implicit def shortParser[F[_]]: ShortParser[F] = new ShortParser[F]()
+  implicit def datePArser[F[_]]: DateParser[F] = new DateParser[F]()
```
Comment (Daenyth) [inline core/src/main/scala/org/http4s/rho/bits/StringParser.scala]: `PA` here probably wants to be `Pa`

**Comment by bryce-anderson** [comment]
Thanks @eklavya!

**Comment by eklavya** [comment]
Thanks @bryce-anderson @jcranky @ChristopherDavenport :)

**Pull request merged**